app/core/middleware.py

```diff
@@ -46,3 +47,33 @@ def middleware(request: Request):
         return get_response(request)
 
     return middleware
+
+
+def downstream_caching(get_response: Callable[[Request], Response]) \
+        -> Callable[[Request], Response]:
+    def middleware(request: Request):
+        public_headers = {
+            'public': True,
+            'max-age': 60 * 5,
+            'stale-while-revalidate': 60 * 5,
+            'stale-if-error': (60 * 60) * 24,
+        }
+
+        private_headers = {
+            'private': True,
+            'max-age': 0,
+            'must-revalidate': True,
+        }
+
+        response = get_response(request)
+
+        if request.META.get(settings.AUTHORIZATION_PREVIEW_HEADER, False):
+            cache_headers = private_headers
```

Comment:
this is a bit magic, if it's someone who can preview then don't cache. Works for `journal--preview`, may be good enough but it's a strange decision

Reply:
For now this part should be fine as its underlying intention is: if you show private content, do not cache it anywhere (especially in public caches)